src/metrics/representation.py

```diff
@@ -146,228 +153,243 @@ def compute(self) -> float:
         Finally, it compute metric.
 
         Returns:
-            Metric value.
+            metric: Metric value.
         """
         vectors = torch.cat(self.vectors).numpy()
-        
         if self.normalize_vectors:
             vectors = normalize(vectors)
 
         if self.dataset_type == DatasetType.CLASSIFICATION:
             # if classification dataset
             targets = torch.cat(self.targets).numpy()
             # prepare data
-            q_vecs, db_vecs, relevants, scores, \
-                db_idxs, q_order_idxs = self.prepare_classification_data(vectors, targets)
+            relevant_idxs, gallery_idxs, query_row_idxs = self.prepare_classification_data(targets)
+            # mock scores and query column indexes because it belong to representation data
+            scores = None
+            query_column_idxs = None
+            # all query vectors is in relevant, so create array with True elements
+            query_as_relevant = np.array([True] * len(gallery_idxs))
```

Comment:
Not consistent with the `prepare_representation_data` call where you return `query_as_relevant` array - I think you should do the same way with `prepare_classification_data`